.github/workflows/azure-e2e-test.yml

```diff
@@ -164,6 +169,7 @@ jobs:
     needs:
     - build-podvm-image
     - build-caa-container-image
+    if: always() && !cancelled()
```

Comment:
Maybe I'm misreading the documentation, but given that `always()` returns true always, do we need it here?

Reply:
Ah, so this is so this job runs even if `build-podvm-image` is skipped, makes sense.

Let's do what you suggest then.